activerecord/lib/active_record/connection_handling.rb

```diff
@@ -160,6 +160,10 @@ def connected_to(database: nil, role: nil, shard: nil, prevent_writes: false, &b
 
         with_handler(role, &blk)
       elsif shard
+        unless role
+          raise ArgumentError, "`connected_to` cannot accept a `shard` argument without a `role`."
+        end
+
         with_shard(shard, role || current_role, prevent_writes, &blk)
```

Comment:
I think we no longer need the `role || current_role` condition here since an error will be raised if `role` is not present.

Reply:
Ah! Good catch, will fix that.